### PR TITLE
github-actions: Enhancements for automatic CRDs update feature

### PR DESCRIPTION
**Pull request by migruiz4**
### Description of the change

Adds several enhancements to the automatic CRDs update workflow to increase its coverage.

The new features are:
   - `# Conditional` - Adds a conditional {{if}}/{{end}} to the downloaded upstream CRD
   - `# VersionOf` - Name of a subcomponent, its version will be used for CRD tracking instead of the main component version
   - `# UseKustomize` - If set to true, uses Kustomize to render the CRDs
   - `# RequiresFilter` - If set to true, uses yq to filter resources having 'kind: CustomResourceDefinition', useful when using 'install.yaml' file as upstream source 

### Benefits

Increase automatic CRD coverage from 7 out of 16, to 15 out of 16. 
The only exception is `cert-manager`, since upstream uses templating that needs to be manually modified.

### Summary of the method used by each chart

- Base feature (Source + Version)
  - bitnami/argo-cd
  - bitnami/argo-workflows
  - bitnami/grafana-operator
  - bitnami/kube-prometheus
  - bitnami/kuberay
  - bitnami/pinniped
  - bitnami/whereabouts

- Base feature + Conditional
  - bitnami/contour
  - bitnami/external-dns
  - bitnami/metallb

- Base feature + Split file
    - bitnami/multus-cni

- Base feature + Subcomponent version
    - bitnami/rabbitmq-cluster-operator

- Base feature + Conditional + Subcomponent version
    - bitnami/flux

- Base feature + Kustomize + Subcomponent version
    - bitnami/apisix
    - bitnami/kong

- Excluded
    - bitnami/cert-manager

### CODE REVIEW

**Review by rafariossaa — DISMISSED** [review]
LGTM

**Comment by migruiz4** [comment]
These changes have been tested here: https://github.com/migruiz4/charts/pulls

File: .github/workflows/ci-pipeline.yml [hunk 1]
```diff
@@ -158,25 +158,74 @@ jobs:
         env:
           CHART: ${{ needs.get-chart.outputs.chart }}
         run: |
-          APP_VERSION="$(yq e '.appVersion' bitnami/${CHART}/Chart.yaml)"
           # Updating CRDs stored at 'bitnami/$CHART/crds' (not 'bitnami/$CHART/templates/crds')
-          mapfile -t crd_files < <(find "bitnami/${CHART}/crds" -name "*.yaml" -o -name "*.yml" 2>/dev/null || true)
+          mapfile -t crd_files < <(find "bitnami/${CHART}/crds" "bitnami/${CHART}/templates/crds" -name "*.yaml" -o -name "*.yml" 2>/dev/null || true)
```
Comment (fmulero) [inline .github/workflows/ci-pipeline.yml:162]: Could you change the comment above?

File: .github/workflows/ci-pipeline.yml [hunk 2]
```diff
@@ -158,25 +158,74 @@ jobs:
...
               crd_version="$(head -n 2 $file | tail -n 1 | grep -E "^# ?Version: ?" | sed -E 's|^# ?Version: ?||' || true)"
               if [[ -z "$crd_version" ]]; then
                 echo "error=CRD file '${file}' does not include the '#Version: <version> header'"
                 exit 1
               fi
+              # Additional headers may be used for extra features
+              # Conditional - Adds a conditional {{if}}/{{end}} to the downloaded upstream CRD
+              # VersionOf - Name of a subcomponent, its version will be used for CRD tracking instead of the main component version
+              # UseKustomize - If set to true, uses Kustomize to render the CRDs
+              # SplitFile - If set to true, uses yq to filter resources having 'kind: CustomResourceDefinition', useful when using 'install.yaml' file as upstream source
```
Comment (fmulero) [inline .github/workflows/ci-pipeline.yml:177]: Not sure if `SplitFile` is the best name here. As I can  see the code is filtering a yml file to select only the CRDs and the result is just one file... Maybe `FilterFile` is a better option.

**Review by fmulero — COMMENTED** [review]
Great @migruiz4!, just two little comments. Could you give it a glance?

File: .github/workflows/ci-pipeline.yml (hunk 2 above)
Comment (migruiz4) [reply .github/workflows/ci-pipeline.yml:177]: Hmm, but the result won't be a single file either, `SplitFile` may yield one or more CRD files when an 'install.yaml' is provided.

**Review by fmulero — APPROVED** [review]
LGTM!

**Pull request merged**